core/types.mli

```diff
@@ -220,6 +219,10 @@ val type_arg_of_quantifier : quantifier -> type_arg
 
 val primary_kind_of_type_arg : type_arg -> PrimaryKind.t
 
+val eq_quantifiers : quantifier -> quantifier -> bool
+
+val add_quantified_vars : quantifier list -> TypeVarSet.t -> TypeVarSet.t
```

Comment:
Not sure how. `TypeVarSet` is just an alias for `Utility.IntSet`, which is an alias for `Set.Make(Int)`. Can I add functions to already instantiated modules?

Reply:
You can do something along the lines of 
```ocaml
module TypeVarSet = struct
include Utility.IntSet

let add_quantifiers : quantifier list -> t -> t
  = fun qs s -> IntSet.( .... )
```

Reply:
Ok, I see. I'll defer doing that until Monday, since that will be a bit more work - I need to first factor out the definitions of `quantifier` and related functions. Probably worth putting them `CommonTypes`.

Reply:
> I need to first factor out the definitions of `quantifier` and related functions. Probably worth putting them `CommonTypes`.

I spent some time doing this and decided that I don't want to follow this direction. I mean, it is generally a Good Thing to do, but doing this properly would require a major refactoring of datatypes, which is well beyond the scope of what I wanted to do in this PR - c.f. #637. Doing this less properly is possible but a waste of time IMO.